saleor/webhook/response_schemas/payment.py

```diff
@@ -99,3 +100,24 @@ class ListStoredPaymentMethodsSchema(BaseModel):
             description="List of stored payment methods.",
         ),
     ]
+
+
+class StoredPaymentMethodDeleteRequestedSchema(BaseModel):
+    result: Annotated[  # type: ignore[name-defined]
+        Literal[
+            StoredPaymentMethodRequestDeleteResult.SUCCESSFULLY_DELETED.name,
+            StoredPaymentMethodRequestDeleteResult.FAILED_TO_DELETE.name,
+            StoredPaymentMethodRequestDeleteResult.FAILED_TO_DELIVER.name,
+        ],
```

Comment:
```suggestion
    result: Annotated[
        StoredPaymentMethodRequestDeleteResult,
        Field(
            description="Result of the request to delete the stored payment method.",
        ),
        BeforeValidator(lower_values),
    ]
```
Why not directly use enum `StoredPaymentMethodRequestDeleteResult` as the type here, if we expect `result` to be of this type later (see other comment)? I see that we are using Enum names for validation and that Pydantic is using Enum values for that - but if, as in our case, names are uppercase versions of values, adding such before validator `BeforeValidator(lower_values)` should do the work. It can be done even better by creating a custom enum type, which uses names for validation as described here: https://github.com/pydantic/pydantic/discussions/2980

If we are not happy with the resulting schema, it can be edited, for example using `WithJsonSchema`

Reply:
I added `EnumByName` annotated type